test/object-store/sync/flx_role_change.cpp

```diff
@@ -857,4 +857,299 @@ TEST_CASE("flx: role changes during bootstrap complete successfully", "[sync][fl
+                                REQUIRE_FALSE(data.error_info->is_fatal);
+                                logger->debug("ROLE CHANGE: role change error received");
+                                role_change_error = true;
+                            }
+                            return std::nullopt;
+                        }
+                        std::optional<ClientResetTestState> new_state = std::nullopt;
+                        if (!update_role_state) {
+                            // If update_role_state is cleared, then no longer need to track the state
+                            return std::nullopt;
+                        }
+                        switch (data.event) {
+                            case Event::BindMessageSent:
+                                // Capture the bind message prior to the client reset error
+                                if (cur_state == ClientResetTestState::start) {
+                                    REQUIRE_FALSE(client_reset_error);
+                                    new_state = ClientResetTestState::bind_before_cr_session;
+                                }
+                                else if (cur_state == ClientResetTestState::cr_session_integrated) {
+                                    REQUIRE(client_reset_error);
+                                    new_state = ClientResetTestState::bind_after_cr_session;
+                                }
+                                break;
+                            case Event::ClientResetMergeComplete:
+                                REQUIRE(cur_state == ClientResetTestState::bind_after_cr_session);
+                                REQUIRE_FALSE(is_fresh_path);
+                                REQUIRE(client_reset_error);
+                                new_state = ClientResetTestState::merged_after_cr_session;
+                                break;
+                            case Event::IdentMessageSent:
+                                // Skip the IDENT message if the client reset error hasn't occurred
+                                if (!client_reset_error)
+                                    break;
+                                // Wait for the ident after the client reset error
+                                if (cur_state == ClientResetTestState::bind_before_cr_session) {
+                                    REQUIRE(is_fresh_path);
+                                    new_state = ClientResetTestState::cr_session_ident;
+                                }
+                                else if (cur_state == ClientResetTestState::merged_after_cr_session) {
+                                    REQUIRE_FALSE(is_fresh_path);
+                                    new_state = ClientResetTestState::ident_after_cr_session;
+                                }
+                                break;
+                            // A bootstrap message was processed by the client reset session
```

Comment:
do we need `REQUIRE(is_fresh_path);`? same for the events below I guess.

Reply:
I don't think it's necessary - I primarily used it for `IdentMessageSent ` and `ClientResetMergeComplete` to verify the new state value and state progression are correct.